### PR TITLE
backend-tests: provides/depends selection only available in "Enterprise"

### PR DESCRIPTION
**Pull request by mchalski**
https://tracker.mender.io/browse/MEN-3312

extending an existing case to prove "os" and "professional" plans will not get the provides/depends selection.

actually, the first 2 commits are bugfixes (missing tools - only provided by mender-qa).

https://gitlab.com/Northern.tech/Mender/mender-qa/pipelines/133235225

(adding @lluiscampos for a look over the first 2 commits)

### CODE REVIEW

File: backend-tests/run [hunk 1]
```diff
@@ -90,9 +92,24 @@ parse_args(){
...
     docker build -t mender-backend-tests-runner -f $INTEGRATION_PATH/backend-tests/docker/Dockerfile $INTEGRATION_PATH/backend-tests
 }
 
+get_runner_requirements() {
+    MENDER_ARTIFACT_BRANCH=$($INTEGRATION_PATH/extra/release_tool.py --version-of mender-artifact)
+    echo "downloading mender-artifact/$MENDER_ARTIFACT_BRANCH"
+
+    curl --fail "https://d1b0l86ne08fsf.cloudfront.net/mender-artifact/${MENDER_ARTIFACT_BRANCH}/linux/mender-artifact" \
+         -o "$TOOLS/mender-artifact" \
+         -z "$TOOLS/mender-artifact"
```
Comment (kacf) [inline backend-tests/run:108]: :+1:

File: tests/run.sh [hunk 2]
```diff
@@ -98,6 +98,22 @@ function get_requirements() {
         exit 1
     fi
 
+    curl --fail "https://raw.githubusercontent.com/mendersoftware/mender/master/support/modules-artifact-gen/directory-artifact-gen" \
+         -o downloaded-tools/directory-artifact-gen \
+         -z downloaded-tools/directory-artifact-gen
```
Comment (kacf) [inline tests/run.sh:103]: I think this should also use

```bash
MENDER_BRANCH=$($INTEGRATION_PATH/extra/release_tool.py --version-of mender)
```

File: backend-tests/tests/test_artifact.py [hunk 3]
```diff
@@ -507,3 +507,53 @@ def test_upload_artifact_selection_match_depends_multiple_artifacts_smallest_siz
...
+            )
+            stdout = p.stdout.read().decode("utf-8")
+            m_size = re.search(r".*size: *([0-9]+).*", stdout, re.M | re.I)
+            assert m_size is not None
+            size = int(m_size.group(1))
+
+            # if provides/depends wasn't ignored - the matching, larger
+            # artifact should have been selected
+            # that's not the case, and we selected 'smallest of all'
+            assert size == 256
```
Comment (kacf) [inline backend-tests/tests/test_artifact.py:557]: Nice test!

File: tests/run.sh (hunk 2 above)
Comment (mchalski) [reply tests/run.sh:103]: fixed

**Pull request merged**